docs/proposal-rose-suite-run.md

```diff
@@ -136,16 +236,28 @@ sources on installation. Other things to consider:
 The `rose suite-run` command calls `cylc validate --strict` by default.
 Automatic suite validation should become the default behaviour for `cylc run`
 and `cylc reload`. Things to consider:
-* Is the *strict* mode necessary by default?
-* `rose suite-run` does not currently validate Rose items. This should be
-  changed to ensure that all aspects of the suite are validated uniformly.
+* `rose suite-run` does not currently validate Rose apps against their
+  metadata. Rose apps should be validated against their metadata by default,
+  but contain switches in the manner of bandit or pylint in a config file.
```

Comment:
Not so sure about this, If the config doesn't fit the metadata, either turn off the metadata or fix the config.

Reply:
I've re-worded this, and hope it fits a little better.

Reply:
I'm not sure switches are a good idea.

Reply:
Why not?

Reply:
> Why not?

1. Because I can't really think of legitimate use-cases.
2. Because people might use them!
3. For one off exceptions we should provide a `--no-validate` option or something of that ilk.

*Longer explanation*

Rose allows people to specify metadata and populate configurations. We then validate the configurations against the metadata to see if it fits. If it doesn't either the configuration is broken and needs fixing or the metadata is out of date and needs fixing.

If we give people switches they can hardcode in their configuration to turn specific checks off is, I think, kinda a bad idea. These switches would be like exceptions and are kinda an extension of the metadata itself but hardcoded in the configuration. It would encourage divergence of metadata and configurations. It's kinda OK when its your little hack because you couldn't be bothered to update the metadata when you were fiddling with some setting, but when that gets committed to the repository someone else will inherit this hack. It might even set the precedent this this is how you get `rose macro -V` to pass.

Reply:
Thank you Oliver! I got this from you in person but I like that  you wrote it up for completeness.